pactflow/client.ts

```diff
@@ -61,16 +88,90 @@ export class PactflowClient implements Client {
       );
     }
 
-    async review(body: RefineInput): Promise<RefineResponse> {
-      // submit review request
+    /**
+     * Resolve the OpenAPI specification from the provided input.
+     * @param remoteOpenAPIDocument The remote OpenAPI document to resolve.
+     * @returns The resolved OpenAPI document.
+     * @throws Error if the resolution fails.
+     */
+    async resolveOpenAPISpec(remoteOpenAPIDocument: RemoteOpenAPIDocument): Promise<OpenAPI> {
+      const openAPISchema = RemoteOpenAPIDocumentSchema.safeParse(remoteOpenAPIDocument);
+      
+      if (openAPISchema.success && remoteOpenAPIDocument) {
+        const unresolvedSpec = await this.getRemoteSpecContents(openAPISchema.data);
+        const resolvedSpec = await Swagger.resolve({ spec: unresolvedSpec });
+
+        if (!resolvedSpec.errors?.length) {
+          return resolvedSpec.spec;
+        }
+
+        throw new Error(`Failed to resolve OpenAPI document: ${resolvedSpec.errors?.join(", ")}`);
+      }
+
+      throw new Error(`Invalid RemoteOpenAPIDocument: ${JSON.stringify(openAPISchema.error?.issues)}`);
```

Comment:
This is a minor nit, but I think it is nicer to have early exits than repeated nested conditions.

E.g.:

```
if (!openAPISchema.success) {
   throw new ...
}

// now we know OpenAPI schema was retrieved.

if (resolvedSpec.errors) {
  throw new ...
}

// Now we know there are no errors
```

I don't code in TypeScript all that often, so what I'm suggesting is contingent on TypeScript's inferences working as I expect.